spyder/plugins/editor/panels/scrollflag.py

```diff
@@ -34,6 +38,23 @@ def __init__(self, editor):
         self._range_indicator_is_visible = False
         self._alt_key_is_down = False
 
+        # Define permanent Qt colors that are needed for painting the flags
+        # and the slider range.
+        self._facecolors = {
```

Comment:
When are these values updated in case of syntax color theme change?

Reply:
Good point. Unless I'm mistaken, only the `occurence_color` is affected by this, so I guess that value could be simply updated in the `_apply_highlighter_color_scheme` method of the codeeditor?

Reply:
hmmm, I think this should be done through a signal instead because the editor seems to never act directly on panels.

Reply:
...or I can put this back in the paintEvent, but this will degrade a little bit the painting performances. @goanpeca What do you think?

Reply:
I woulf move the creation of those helper variables to a separate method and see what signal method could be attached to panels 🤷🏻‍♂️

Reply:
> 
> 
> I woulf move the creation of those helper variables to a separate method and see what signal method could be attached to panels 🤷🏻‍♂️

Yes, I agree with this. We will need to add a new signal for this I think.

Reply:
The signal should probably be related to the theme color which is more “general” and could be used by other panels.

Reply:
What do you mean concretely by that? Are you asking to just change the name of the signal I added or does this imply anything else?

Reply:
Yes the signal name :-)

Reply:
Ok thank you! :)